packages/store/src/decorators/select/select.ts

```diff
@@ -23,9 +24,15 @@ export function Select<T>(rawSelector?: T, ...paths: string[]): PropertyDecorato
         enumerable: true,
         configurable: true,
         get(): PropertyType<T> {
+          // The `localInject` will be tree-shaken away in apps that
+          // still use the View Engine.
+          const store = ɵivyEnabled ? localInject(this, Store) : null;
```

Comment:
what do you think about put to `store/internals/utils.ts`

```ts
export function injectNgxsStore(): Store | null {
  return ɵivyEnabled ? localInject(this, Store) : null;
}

export function injectNgxsConfig(): NgxsConfig | null {
  return ɵivyEnabled ? localInject(this, NgxsConfig) : null;
}
```

and as 

```ts
// ...
get(): PropertyType<T> { 
 
   if (isNil(this[selectorId])) {
      selector = selector || createSelectorFn(injectNgxsConfig(), name, rawSelector, paths);
      this[selectorId] = createSelectObservable(selector, injectNgxsStore());
   }

  return this[selectorId];
}
```

Reply:
This seems to be an over-engineering.

Reply:
up to you